src/main/scala/com/gu/scanamo/DerivedDynamoFormat.scala

```diff
@@ -55,3 +56,23 @@ trait DerivedDynamoFormat {
       def write(t: T): AttributeValue = formatR.value.write(gen.to(t))
     }
 }
+
+trait NoFormatError {
+  import scala.language.experimental.macros
+  implicit def moreHelpfulErrorMessage[T]: DynamoFormat[T] = macro MacroOfLastResort.moreHelpfulErrorMessage[T]
+}
+
+@bundle
+class MacroOfLastResort(val c: scala.reflect.macros.blackbox.Context) {
```

Comment:
It can also be thought of as your entry point to the reflection API. Importing `c.universe._` brings into scope everything your macro will need, including:

* quasiquotes (`q"..."`)
* all the AST types such as `TermName`, `ClassDef`, `Apply`, `Select`
* methods for looking up/manipulating types, e.g. `weakTypeOf`, `appliedType`, `lub`
* Useful debugging methods such as `showRaw` and `showCode`